src/object.c

```diff
@@ -736,7 +743,7 @@ size_t objectComputeSize(robj *o, size_t sample_size) {
             asize = sizeof(*o)+sizeof(dict)+(sizeof(struct dictEntry*)*dictSlots(d));
             while((de = dictNext(di)) != NULL && samples < sample_size) {
                 ele = dictGetKey(de);
-                elesize += sizeof(struct dictEntry) + sdsAllocSize(ele);
+                elesize += sizeOfStringObject(ele);
```

Comment:
After the unlink changes, ele is a sds, not an robj*. So, this does not look correct. I think it should be 
elesize += sdsAllocSize(ele);
I agree with the removal of sizeof(struct dictEntry) as it is already counted above.